File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0
        r=0
        d={}
        ans=0
        while(r<len(s)):
            if(s[r] not in d):
                d[s[r]]=1
                r+=1
                ans=max(ans,r-l)
            else:
                while(s[r]!=s[l]):
                    del d[s[l]]
                    l+=1
                del d[s[l]]
                l+=1
        ans=max(ans,r-l)
        return ans

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        int n = s.Length;
        HashSet<char> set = new HashSet<char>();
        int maxLength = 0, i = 0, j = 0;
        while (i < n && j < n)
        {
            if (!set.Contains(s[j]))
            {
                set.Add(s[j++]);
                maxLength = Math.Max(maxLength, j - i);
            }
            else
            {
                set.Remove(s[i++]);
            }
        }
        return maxLength;
    }
}